syntax = "proto3";


service Chat{
    rpc CreateAccount(Account) returns (ServerResponse) {}
    rpc Login(Account) returns (ServerResponse) {}
    rpc ListAccounts(NoParam) returns (Accounts) {}
    rpc SendMessage(MessageInfo) returns (ServerResponse) {}
}

message NoParam {
}

message Account {
    string name = 1;
}

message Accounts {
    string accounts = 1;
}

message ServerResponse {
    string message = 1;
    bool received = 2;
}

message MessageInfo {
    Account destination = 1;
    Account source = 2;
    string text = 3;
}